### PR TITLE
Use a set to store tagOpen completions

### LINKED ISSUE #141: Duplicate dependency completion entries
Still trying to get lemminx-maven working in vscode.
Now I'm at a point where I get completion for dependencies, but I see duplicates for some of them:

Screen Shot 2020-09-08 at 7 22 21 PM

### PR DESCRIPTION
**Pull request by AObuchow**
Try to prevent duplicate completions
Fix #141

Signed-off-by: Andrew Obuchowicz <andrew@aobuchow.com>

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java [hunk 1]
```diff
@@ -328,11 +329,12 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
 					.stream()
 					.map(artifacts -> Collections.max(artifacts, highestVersionWithDescriptionComparator))
 					.map(artifactInfo -> toGAVCompletionItem(artifactInfo, request, gavInsertionStrategy))
-					.forEach(response::addCompletionItem);
+					.forEach(completionItems::add);
 		}
 		if (request.getNode().isText() && (allArtifactInfos.isEmpty() || request.getNode().getTextContent().contains("$"))) {
 			completeProperties(request).forEach(response::addCompletionAttribute);
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:335]: I'm not sure what's the difference between a completion item and completion attribute? Maybe the properties should also be added to the set of completionItems.

**Comment by mickaelistria** [comment]
Please add a test case beforehand for that issue, it will make things easier.

**Comment by AObuchow** [comment]
> Please add a test case beforehand for that issue, it will make things easier.

I'm not entirely sure yet how to reproduce the bug as my suspicion is that the local repo & index were providing the same completion. Perhaps I can setup a test to download an artifact from the soffid repo (on hover) then trigger a completion to see if duplicate completions appear (one from local repo and one from index)

**Comment by mickaelistria** [comment]
Yes, I also believe that local repo + index containing the artifact is the scenario. However, I don't think we should pick one or the other as in some cases, one or the other may have more data available. We should instead try to "reduce" multiple completion items for the same GAV to a single one, merging their input if its different.

**Comment by AObuchow** [comment]
@mickaelistria ok good idea. So I'll try to come up with a function that merges GAV completion items to the one that has the most data available then.

**Comment by mickaelistria** [comment]
This is too outdated, let's close.

**Pull request closed without merging**